lib/chef_zero/endpoints/organization_policy_groups_endpoint.rb

```diff
@@ -0,0 +1,262 @@
+            end
+
+            return json_response(200, response_data)
+          end    # if policy_groups.size > 0
+        end    # end /policy_groups
+
+        # /policy_groups/{policy_group}
+        if request.rest_path.last(2).first == "policy_groups"
+          data = {
+            uri: build_uri(request.base_uri, request.rest_path),
+            policies: get_policy_group_policies(request)
+          }
+          return json_response(200, data)
+        end
+
+        # /policy_groups/{policy_group}/policies/{policy_name}
+        if request.rest_path.last(2).first == "policies"
+          uri_params = fetch_uri_params(request)
+
+          # fetch /organizations/{organization}/policies/{policy_name}/revisions/{revision_id}
+          revision_id = parse_json(get_data(request))
+          result = get_data(request, ["organizations", uri_params[:org_name], "policies",
+                            uri_params[:policy_name], "revisions", revision_id], :nil)
+          return already_json_response(200, result)
+        end
+      end    # end get()
+
+      # Create or update the policy document for the given policy group and policy name. If no policy group
+      # with the given name exists, it will be created. If no policy with the given revision_id exists, it
+      # will be created from the document in the request body. If a policy with that revision_id exists, the
+      # Chef Server simply associates that revision id with the given policy group. When successful, the
+      # document that was created or updated is returned.
+
+      # build a hash of {"some_policy_name"=>{"revision_id"=>"909c26701e291510eacdc6c06d626b9fa5350d25"}}
+      def get_policy_group_policies(request)
+        policies_revisions = {}
+
+        policies_path = request.rest_path + ["policies"]
+        policy_names = list_data(request, policies_path)
+        policy_names.each do |policy_name|
+          revision = parse_json(get_data(request, policies_path + [policy_name]))
+          policies_revisions[policy_name] = { revision_id: revision}
+        end
+
+        policies_revisions
+      end
+
+      def set_opts
+        [ :create_dir, :data_store_exceptions ]
+      end
+
+
+      ## MANDATORY FIELDS AND FORMATS
+      # * `revision_id`: String; Must be < 255 chars, matches /^[\-[:alnum:]_\.\:]+$/
+      # * `name`: String; Must match name in URI; Must be <= 255 chars, matches /^[\-[:alnum:]_\.\:]+$/
+      # * `run_list`: Array
+      # * `run_list[i]`: Fully Qualified Recipe Run List Item
+      # * `cookbook_locks`: JSON Object
+      # * `cookbook_locks(key)`: CookbookName
+      # * `cookbook_locks[item]`: JSON Object, mandatory keys: "identifier", "dotted_decimal_identifier"
+      # * `cookbook_locks[item]["identifier"]`: varchar(255) ?
+      # * `cookbook_locks[item]["dotted_decimal_identifier"]` ChefCompatibleVersionNumber
+
+
+      def cookbook_locks_valid?(cookbook_locks)
```

Comment:
Yes: the validations were actually written while @danielsdeleo was simultaneously looking at erchef, just to speed things up.